pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -671,6 +796,10 @@ def expval(self, observable, shot_range=None, bin_size=None):
             # use specialized functors to compute expval(Hermitian)
             if observable.name == "Hermitian":
                 observable_wires = self.map_wires(observable.wires)
+                if len(observable_wires) > self._num_local_wires and self._mpi:
+                    raise RuntimeError(
+                        "MPI backend does not support Hermitian with number of target wires larger than local wire number."
+                    )
```

Comment:
```suggestion
                if self._mpi and len(observable_wires) > self._num_local_wires:
                    raise RuntimeError(
                        "MPI backend does not support Hermitian with number of target wires larger than local wire number."
                    )
```

Since we only do the comparison if the above boolean is enabled